### PR TITLE
routeguide: reimplement distance calculation

### PR DESCRIPTION
**Pull request by jtattermusch**
Same as https://github.com/grpc/grpc/pull/14794.

### CODE REVIEW

**Comment by ejona86** [comment]
Making a comment to try to kick cla/linuxfoundation to do its thing.

File: examples/src/main/java/io/grpc/examples/routeguide/RouteGuideServer.java [hunk 1]
```diff
@@ -268,25 +268,23 @@ private Feature checkFeature(Point location) {
...
+      int r = 6371000; // earth radius in meters
+      double lat1 = toRadians(RouteGuideUtil.getLatitude(start));
+      double lat2 = toRadians(RouteGuideUtil.getLatitude(end));
+      double lon1 = toRadians(RouteGuideUtil.getLongitude(start));
+      double lon2 = toRadians(RouteGuideUtil.getLongitude(end));
+      double deltaLat = lat2 - lat1;
+      double deltaLon = lon2 - lon1;
+
+      double a = sin(deltaLat / 2) * sin(deltaLat / 2)
+          + cos(lat1) * cos(lat2) * sin(deltaLon / 2) * sin(deltaLon / 2);
```
Comment (ejona86) [inline examples/src/main/java/io/grpc/examples/routeguide/RouteGuideServer.java]: Like the other implementation, this looks wrong, as we're passing non-radians to cos.

Comment (jtattermusch) [reply examples/src/main/java/io/grpc/examples/routeguide/RouteGuideServer.java]: Not sure what you mean:
`double lat1 = toRadians(RouteGuideUtil.getLatitude(start));`

Comment (ejona86) [reply examples/src/main/java/io/grpc/examples/routeguide/RouteGuideServer.java]: nm

**Pull request merged**